src/compiler/factory.ts

```diff
@@ -1595,6 +1606,321 @@ namespace ts {
         return createVoid(createLiteral(0));
     }
 
+    export function createExportDefault(expression: Expression) {
+        return createExportAssignment(/*decorators*/ undefined, /*modifiers*/ undefined, /*isExportEquals*/ false, expression);
+    }
+
+    export function createExternalModuleExport(exportName: Identifier) {
+        return createExportDeclaration(/*decorators*/ undefined, /*modifiers*/ undefined, createNamedExports([createExportSpecifier(exportName)]));
+    }
+
+    // Utilities
+
+    function asName(name: string): Identifier;
+    function asName(name: Identifier): Identifier;

```

Comment:
I think `string | Identifier` covers both the `string` and `Identifier` overloads. Are they just there because they are easier to understand for people who don't know about union types?